markymark/Classes/Styling/Protocols/BaseFontStylingRule.swift

```diff
@@ -22,33 +22,27 @@ extension ItemStyling {
     }
 
     func neededFont() -> UIFont? {
+        
+        var font: UIFont? = neededBaseFont()
 
-        if var font: UIFont? = neededBaseFont() {
-
-            if shouldFontBeBold() {
-
-                font = font?.makeBold()
-            }
-
-            if shouldFontBeItalic() {
-
-                font = font?.makeItalic()
-            }
+        if shouldFontBeBold() {
+            font = font?.makeBold()
+        }
 
-            if shouldFontBeBold() && shouldFontBeItalic() {
-                font = font?.makeItalicBold()
-            }
+        if shouldFontBeItalic() {
+            font = font?.makeItalic()
+        }
 
-            if let textSize = neededTextSize() {
+        if shouldFontBeBold() && shouldFontBeItalic() {
```

Comment:
I don't like how bold and italics don't work if added separately but I can understand it is out of scope for this review.